components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/util/RestApiUtil.java

```diff
@@ -526,4 +534,68 @@ public static ZoneId getRequestTimeZone(String timestamp) {
         return ZonedDateTime.parse(timestamp).getZone();
     }
 
+    /**
+     * Get defined HTTP methods in the swagger definition as space separated string
+     *
+     * @param request           Request
+     * @param serviceMethodInfo Method informations for the request
+     * @return Http Methods as a space separated string
+     * @throws APIManagementException if failed to get defined http methods
+     */
+    public static synchronized String getDefinedMethodHeadersInSwaggerContent(
```

Comment:
Do we need to make this synchronized?

Reply:
Thank you. Resolved with a static block. Parsing swagger definition is now done when the class is loaded.
fix: https://github.com/wso2/carbon-apimgt/pull/5095/commits/ac2a2a0304d573e35d3a8105350692f0beb36118